### PR TITLE
TCI for DG and FD grid for advection system

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

- [x] ~~Dependent on #3494~~

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+template <size_t Dim>
+bool TciOnFdGrid<Dim>::apply(const Scalar<DataVector>& dg_u,
+                             const Mesh<Dim>& dg_mesh,
+                             const double persson_exponent) noexcept {
+  constexpr double persson_tci_epsilon = 1.0e-18;
+  // In order to prevent rapid switching between the DG and FD schemes, we use
+  // persson exponent += 1 for the TCI when deciding whether to switch back to
+  // DG.
+  return ::evolution::dg::subcell::persson_tci(
+      dg_u, dg_mesh, persson_exponent + 1.0, persson_tci_epsilon);
```
Comment (yoonso0-0) [inline src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.cpp]: @nilsdeppe As I understood from the paper, it seems that the `persson_exponent` needs to be +=1 here before switching to DG from FD (from the statements in the comments above).. is that correct?

Comment (nilsdeppe) [reply src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.cpp]: The correct exponent is passed into the function, you don't need to change it here. That is, it's passed in with the `+1` included

Comment (yoonso0-0) [reply src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.cpp]: Yep, I found the action doing it. Fixed :)

File: src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/// \endcond
+
+namespace ScalarAdvection::subcell {
+/*!
+ * \brief Troubled-cell indicator applied to the finite difference subcell
+ * solution to check if the corresponding DG solution is admissible.
+ *
+ * Applies the Persson TCI to \f$U\f$ on the DG grid.
+ *
+ * \note TCI is run after reconstructing solution to the DG grid during the
```
Comment (nilsdeppe) [inline src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.hpp]: when you rebase this after the other PR is merged can you update `reconstructing solution` to `reconstructing the solution`?

**Review by nilsdeppe — COMMENTED** [review]
LGTM, just one comment you can squash when rebasing after the ID TCI is merged :)

File: src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.hpp (hunk 2 above)
Comment (nilsdeppe) [reply src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.hpp]: I'm not seeing this. Did you forget to push this change maybe?

Comment (yoonso0-0) [reply src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.hpp]: @nilsdeppe My bad :(

Comment (nilsdeppe) [reply src/Evolution/Systems/ScalarAdvection/Subcell/TciOnFdGrid.hpp]: Oh I've done that a lot :) Just wanted to check :)

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**